### PR TITLE
feat: tooltip component

### PR DESCRIPTION
**Pull request by Arthurmqz**
This update addresses the need for a customizable and reusable set of a tooltip component in the project.

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: packages/components/tooltip/src/hooks/__tests__/useClickOutside.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+    renderHook(() => useClickOutside(ref, handler));
+
+    document.body.appendChild(ref.current);
+
+    fireEvent.mouseDown(document.body);
+
+    expect(handler).toHaveBeenCalledTimes(1);
+
+    document.body.removeChild(ref.current);
```
Comment (joaoprcosta) [inline packages/components/tooltip/src/hooks/__tests__/useClickOutside.spec.tsx:18]: This action could be set in 'afterAll' function outside each test at the beginning.

```suggestion
    afterAll(() => document.body.removeChild(ref.current);)
```

File: packages/components/tooltip/src/Tooltip.tsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+      {...handleEvents({ trigger, showTooltip, hideTooltip, toggleTooltip })}
+    >
+      <span>{childrenLabel}</span>
+      {visible && (
+        <TooltipStyled
+          ref={tooltipRef}
+          styled={mergedThemeValues as Required<TooltipStyledProps>}
+          position={position}
+          isFixedBottom={isFixedBottom}
+          style={{ top: `${coords.top}px`, left: `${coords.left}px` }}
```
Comment (jessywork) [inline packages/components/tooltip/src/Tooltip.tsx:72]: you can pass it to the styles file

File: packages/components/tooltip/src/tooltip.styles.ts [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+  styled: Required<TooltipStyledProps>,
+  isFixedBottom: boolean,
+) => {
+  if (isFixedBottom) return '';
+
+  switch (position) {
+    case 'top':
+      return `
+        &::after {
+          bottom: -5px; /* Flecha abaixo do tooltip */
```
Comment (jessywork) [inline packages/components/tooltip/src/tooltip.styles.ts:15]: remove comment

File: packages/components/tooltip/src/tooltip.styles.ts [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+          left: 50%;
+          transform: translateX(-50%);
+          border-width: 5px 5px 0 5px;
+          border-top-color: ${styled.backgroundColor};
+        }
+      `;
+    case 'bottom':
+      return `
+        &::after {
+          top: -5px; /* Flecha acima do tooltip */
```
Comment (jessywork) [inline packages/components/tooltip/src/tooltip.styles.ts:25]: remove comment

File: packages/components/tooltip/src/tooltip.styles.ts [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+          transform: translateX(-50%);
+          border-width: 0 5px 5px 5px;
+          border-bottom-color: ${styled.backgroundColor};
+        }
+      `;
+    case 'left':
+      return `
+        &::after {
+          top: 50%;
+          right: -5px; /* Flecha à esquerda do tooltip */
```
Comment (jessywork) [inline packages/components/tooltip/src/tooltip.styles.ts:36]: remove comment

File: packages/components/tooltip/src/tooltip.styles.ts [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+          transform: translateY(-50%);
+          border-width: 5px 0 5px 5px;
+          border-left-color: ${styled.backgroundColor};
+        }
+      `;
+    case 'right':
+      return `
+        &::after {
+          top: 50%;
+          left: -5px; /* Flecha à direita do tooltip */
```
Comment (jessywork) [inline packages/components/tooltip/src/tooltip.styles.ts:46]: remove comment

**Review by jessywork — CHANGES_REQUESTED** [review]

**Review by jessywork — APPROVED** [review]

**Review by JuniorDiasOliveira — APPROVED** [review]

**Pull request merged**